src/Microsoft.ML.TimeSeries/Forecast.cs

```diff
@@ -0,0 +1,93 @@
+        /// </summary>
+        /// <param name="dataView">Reference to the <see cref="IDataView"/></param>
+        void Train(IDataView dataView);
+
+        /// <summary>
+        /// Update a forecasting model with the new observations in the form of an <see cref="IDataView"/>.
+        /// </summary>
+        /// <param name="dataView">Reference to the observations as an <see cref="IDataView"/></param>
+        /// <param name="inputColumnName">Name of the input column to update from.</param>
+        void Update(IDataView dataView, string inputColumnName = null);
+
+        /// <summary>
+        /// Perform forecasting until a particular <paramref name="horizon"/>.
+        /// </summary>
+        /// <param name="horizon">Number of values to forecast.</param>
+        /// <returns>Forecasted values.</returns>
+        T[] Forecast(int horizon);
+
+        /// <summary>
+        /// Perform forecasting until a particular <paramref name="horizon"/> and also computes confidence intervals.
+        /// </summary>
+        /// <param name="horizon">Number of values to forecast.</param>
+        /// <param name="forecast">Forecasted values</param>
+        /// <param name="confidenceIntervalLowerBounds">Lower bound confidence intervals of forecasted values.</param>
+        /// <param name="confidenceIntervalUpperBounds">Upper bound confidence intervals of forecasted values.</param>
+        /// <param name="confidenceLevel">Confidence level.</param>
+        void ForecastWithConfidenceIntervals(int horizon, out T[] forecast, out float[] confidenceIntervalLowerBounds, out float[] confidenceIntervalUpperBounds, float confidenceLevel = 0.95f);
+    }
+
+    public static class ForecastExtensions
+    {
+        /// <summary>
+        /// Load a forecasting model.
+        /// </summary>
+        /// <typeparam name="T">The type of <see cref="ICanForecast{T}"/>, usually float. </typeparam>
+        /// <param name="catalog"><see cref="ModelOperationsCatalog"/></param>
+        /// <param name="filePath">File path to save the model to.</param>
+        /// <returns></returns>
+        public static ICanForecast<T> LoadForecastingModel<T>(this ModelOperationsCatalog catalog, string filePath)
+        {
+            var env = CatalogUtils.GetEnvironment(catalog);
+            using (var file = File.OpenRead(filePath))
+            {
+                using (var rep = RepositoryReader.Open(file, env))
+                {
+                    ModelLoadContext.LoadModel<ICanForecast<T>, SignatureLoadModel>(env, out var model, rep, LoaderSignature);
+                    return model;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Save a forecasting model.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="catalog"><see cref="ModelOperationsCatalog"/></param>
+        /// <param name="model">Model to save.</param>
```

Comment:
>Model [](start = 32, length = 5)

Please specify the type of the model by using, for example, <see name="ICanForecast<T>">.